src/js/productDetails.js

```diff
@@ -41,18 +47,38 @@ export default class ProductDetails {
     this.cart = getLocalStorage("so-cart");
     var cartImg = document.querySelector(".cart");
     cartImg.classList.add("anim-out");
-        setTimeout(() => {
-          cartImg.classList.remove("anim-out");
-        }, 300);
-    //console.log("00", this.cart)
+    setTimeout(() => {
+      cartImg.classList.remove("anim-out");
+    }, 300);
+
+    //If the cart isn't empty, we can use .push to add the item
     if (this.cart != null) {
-      this.cart.push(this.product);
-      //console.log("0" , this.cart);
-    } else {
-      //this.cart.push(this.product);
+      
+      //Search for duplicate items
+      var duplicate = this.cart.find(product => this.product.Id == product.Id)
```

Comment:
@avieeich I think this code I added here will help you with your Trello card. All products now have a "Quantity" field. This is how I adjusted it when people add items to the cart, I imagine it will be similar within the cart itself.